### PR TITLE
10608: Properly cleanup HTTPSEverywhereService.

### PR DESCRIPTION
**Pull request by iefremov**
Resolves https://github.com/brave/brave-browser/issues/10608

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_shields/browser/ad_block_base_service.cc [hunk 1]
```diff
@@ -110,10 +110,6 @@ AdBlockBaseService::AdBlockBaseService(BraveComponent::Delegate* delegate)
       weak_factory_(this) {}
 
 AdBlockBaseService::~AdBlockBaseService() {
-  Cleanup();
```
Comment (iefremov) [inline components/brave_shields/browser/ad_block_base_service.cc:113]: Calling virtual functions in destructors is generally a bad practice, and also `Cleanup` is not actually used

File: components/brave_shields/browser/ad_block_regional_service_manager.cc [hunk 2]
```diff
@@ -116,13 +116,6 @@ bool AdBlockRegionalServiceManager::Start() {
   return true;
 }
 
-void AdBlockRegionalServiceManager::Stop() {
```
Comment (iefremov) [inline components/brave_shields/browser/ad_block_regional_service_manager.cc:119]: not used

File: components/brave_shields/browser/ad_block_regional_service_manager.cc [hunk 3]
```diff
@@ -177,7 +170,6 @@ void AdBlockRegionalServiceManager::EnableFilterList(const std::string& uuid,
           std::make_pair(uuid, std::move(regional_service)));
     } else {
       DCHECK(it != regional_services_.end());
-      it->second->Stop();
```
Comment (iefremov) [inline components/brave_shields/browser/ad_block_regional_service_manager.cc:180]: we are deleting `it->second` two lines below, so this is not needed

File: components/brave_shields/browser/base_brave_shields_service.cc [hunk 4]
```diff
@@ -49,12 +49,6 @@ bool BaseBraveShieldsService::Start() {
   return false;
 }
 
-void BaseBraveShieldsService::Stop() {
```
Comment (iefremov) [inline components/brave_shields/browser/base_brave_shields_service.cc:52]: not used outside `ad_block_regional_service_manager.cc` where actually not needed

File: components/brave_shields/browser/https_everywhere_service.cc [hunk 5]
```diff
@@ -105,14 +105,7 @@ HTTPSEverywhereService::HTTPSEverywhereService(
...
 
 HTTPSEverywhereService::~HTTPSEverywhereService() {
-  Cleanup();
-}
-
-void HTTPSEverywhereService::Cleanup() {
-  GetTaskRunner()->PostTask(
-      FROM_HERE,
-      base::Bind(&HTTPSEverywhereService::CloseDatabase,
-                 AsWeakPtr()));
```
Comment (iefremov) [inline components/brave_shields/browser/https_everywhere_service.cc:115]: `AsWeakPtr` would be empty when `Cleanup` is called from destructor, so this is noop

File: components/brave_shields/browser/https_everywhere_service.cc [hunk 6]
```diff
@@ -142,7 +135,6 @@ void HTTPSEverywhereService::InitDB(const base::FilePath& install_dir) {
                         unzipped_level_db_path.AsUTF8Unsafe(),
                         &level_db_);
   if (!status.ok() || !level_db_) {
-    level_db_ = nullptr;
```
Comment (iefremov) [inline components/brave_shields/browser/https_everywhere_service.cc:145]: `level_db_` is cleaned up just below

File: components/brave_shields/browser/https_everywhere_service.cc [hunk 7]
```diff
@@ -105,14 +105,7 @@ HTTPSEverywhereService::HTTPSEverywhereService(
...
 HTTPSEverywhereService::~HTTPSEverywhereService() {
-  Cleanup();
-}
-
-void HTTPSEverywhereService::Cleanup() {
-  GetTaskRunner()->PostTask(
-      FROM_HERE,
-      base::Bind(&HTTPSEverywhereService::CloseDatabase,
-                 AsWeakPtr()));
+  GetTaskRunner()->DeleteSoon(FROM_HERE, level_db_);
```
Comment (simonhong) [inline components/brave_shields/browser/https_everywhere_service.cc:108]: Q) Is there a reason why not delete db here? Or just delaying time consuming tasks?

Comment (iefremov) [reply components/brave_shields/browser/https_everywhere_service.cc:108]: I assume it must be deleted on the background task runner, not on the UI thread

Comment (simonhong) [reply components/brave_shields/browser/https_everywhere_service.cc:108]: I see!

**Pull request merged**